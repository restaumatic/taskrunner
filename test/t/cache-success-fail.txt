# no toplevel
# s3

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir a

(
  cd a
  echo foo > input.txt
  git init -q
  git add input.txt
)

cp -r a b

go() {
  # Note: we want separate workdir for separate repos
  TASKRUNNER_STATE_DIRECTORY="$(pwd)" taskrunner -n mytask bash -e -c '
    snapshot input.txt --cache-success
    echo "Expensive computation"
    false # This command will fail, simulating a task failure
  '
}

(
  cd a
  go || echo "First run failed as expected"
)

(
  cd b
  go # note: should run "expensive computation" again due to previous failure
)
