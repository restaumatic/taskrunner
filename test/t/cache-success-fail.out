-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] error | Failed, exit code: 1
First run failed as expected
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] error | Failed, exit code: 1
-- exit code: 1
